c/polyjuice.h

```diff
@@ -701,6 +701,41 @@ void emit_log(struct evmc_host_context* context, const evmc_address* address,
   return;
 }
 
+/**
+ * check address collision
+ * check existence of eth_addr
+ * - if exists, then check nonce > 0 or it's NOT an EOA: return AddressCollision error
+ * - else, we can overwrite addr mapping
+ * @param overwrite true if there is a collision but we can continue to create a new account
+ * @return 0 means success
+*/
+int check_address_collision(gw_context_t* ctx, const uint8_t eth_addr[ETH_ADDRESS_LEN], bool* overwrite) {
+  gw_reg_addr_t addr = new_reg_addr(eth_addr);
+  uint8_t script_hash[32] = {0};
+  int ret = ctx->sys_get_script_hash_by_registry_address(ctx, &addr, script_hash);
+  if (ret != 0) {
```

Comment:
Use the specific error `GW_ERROR_NOT_FOUND ` to check existence.

https://github.com/nervosnetwork/godwoken-scripts/blob/03011e0933ef739b96eb490b26083e8a3708b019/c/common.h#L320